File: ontrack-extension-scm/src/main/java/net/nemerosa/ontrack/extension/scm/model/SCMFileChangeFilter.java
package net.nemerosa.ontrack.extension.scm.model;

import lombok.Data;

import java.util.List;

/**
 * Defines a filter on the file changes.
 */
@Data
public class SCMFileChangeFilter {

    /**
     * Name of the filter
     */
    private final String name;

    /**
     * List of ANT-like patterns for the paths
     */
    private final List<String> patterns;

}

File: ontrack-extension-scm/src/main/java/net/nemerosa/ontrack/extension/scm/model/SCMFileChangeFilter.kt
package net.nemerosa.ontrack.extension.scm.model

/**
 * Defines a filter on the file changes.
 */
data class SCMFileChangeFilter(
    /**
     * Name of the filter
     */
    val name: String,
    /**
     * List of ANT-like patterns for the paths
     */
    val patterns: List<String>,
)